### DISCUSSION TITLE
send a message to a specific peer

### DISCUSSION THREAD

**Discussion by RemonAhammad** [libp2p/rust-libp2p#4313] [Q&A] [answered]
suppose I have connected with many peers in a swarm. I wanna send my message to a specific one or multiple (specific peers). how can I do that?
I have searched on Google and AI. but not got any proper solution. Thanks.

**Comment by mxinden** [comment]
You can use `libp2p-request-response`. See e.g. the JSON `Codec`:

https://github.com/libp2p/rust-libp2p/blob/master/protocols/request-response/src/json.rs

**Reply by RemonAhammad** [reply]
I am not understanding how can I bind the swarm with Codec. could you give me any code snippets, please?  thanks.
I am a beginner in rust.

**Reply by RemonAhammad** [reply]
could you give a solution?

**Reply by mxinden** [reply, marked as answer]
You can find a code snipped with `Swarm` in https://github.com/libp2p/rust-libp2p/blob/master/protocols/request-response/tests/ping.rs.

**Comment by kehiy** [comment]
hi @blackshadowsoftwareltd,

See [this](https://docs.rs/libp2p/latest/libp2p/request_response/trait.Codec.html#tymethod.read_response) part of document.
The `RequestResponse` can do it. (from docs and the `rust-libp2p` exact code)

**Reply by RemonAhammad** [reply]
could you share your swarm config and behavior?
basically,  this method does not exist in my swarm.
thanks

**Reply by kehiy** [reply]
@blackshadowsoftwareltd , check the message again I update it with a document link.

**Reply by RemonAhammad** [reply]
Thank you so much for your opinion. I have already used Request Response behavior to send a specific peer.
